### PR TITLE
Initial schema for album / photo metadata.

### PR DESCRIPTION
**Pull request by maxpoletto**

### CODE REVIEW

File: internal/server/schema.sql [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+------ Folders
+CREATE TABLE folders (
+    id INTEGER PRIMARY KEY ASC,
+    name TEXT NOT NULL,
+    path TEXT NOT NULL UNIQUE
+);
+CREATE UNIQUE INDEX folders_path ON folders(path);
+
+CREATE TABLE folders_tree (
```
Comment (thomergil) [inline internal/server/schema.sql]: Couldn't we represent the tree structure in the `folders` table itself by giving it a parent_id? Is there a reason you do this in a separate table?

File: internal/server/schema.sql
Comment (thomergil) [inline internal/server/schema.sql:1]: This schema is lacking `ON DELETE` rules.

File: internal/server/schema.sql [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    FOREIGN KEY(album_id) REFERENCES albums(id)
+);
+CREATE INDEX album_aliases_album_id ON album_aliases(album_id);
+
+-- Media (photos / videos).
+-- Describes a logical photo, not a specific file (may have multiple resolutions).
+CREATE TABLE media (
+    id INTEGER PRIMARY KEY,
+    album_id INTEGER NOT NULL,
+    is_photo INTEGER NOT NULL, -- 0: video, 1: photo
```
Comment (thomergil) [inline internal/server/schema.sql]: Might we be better off making this a `media_type` as an enumerable? Who knows what else we may want to store.

File: internal/server/schema.sql [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    display_name TEXT NOT NULL,
+    source_filename TEXT NOT NULL,
+    mtime INTEGER NOT NULL,
+    -- EXIF data
+    exif_time INTEGER,
+    latitude REAL,
+    longitude REAL,
+    camera TEXT,
```
Comment (thomergil) [inline internal/server/schema.sql:60]: Is this coming straight from EXIF? I assume defining cameras (like, bikes in Strava) is beyond the scope, yeah?

File: internal/server/schema.sql [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    exposure_time REAL,
+    aperture REAL,
+    iso INTEGER,
+    flash INTEGER,
+    orientation INTEGER, -- 0: landscape, 1: portrait
+    FOREIGN KEY(album_id) REFERENCES albums(id)
+);
+CREATE INDEX media_album_id ON media(album_id);
+
+CREATE TABLE media_text (
```
Comment (thomergil) [inline internal/server/schema.sql:72]: Can one `media` have multiple `media_text`s? What does that mean? If there's a 1:1 relationship, shouldn't these just be part of `media`?

File: internal/server/schema.sql (hunk 3 above)
Comment (maxpoletto) [reply internal/server/schema.sql:60]: Correct, straight from EXIF (if any)

File: internal/server/schema.sql (hunk 1 above)
Comment (maxpoletto) [reply internal/server/schema.sql]: You're right. I was thinking of it in terms of a downward traversal (from parent to child), but of course that could be done just as efficiently with an index on parent_id. Will change.

File: internal/server/schema.sql (hunk 2 above)
Comment (maxpoletto) [reply internal/server/schema.sql]: OK

File: internal/server/schema.sql (hunk 4 above)
Comment (maxpoletto) [reply internal/server/schema.sql:72]: Yes, multiple media texts depending on language_code.

File: internal/server/schema.sql
Comment (maxpoletto) [reply internal/server/schema.sql:1]: Yes, definitely not finished. Adding some now.

File: internal/server/schema.sql (hunk 4 above)
Comment (thomergil) [reply internal/server/schema.sql:72]: Oh duh.

**Pull request merged**